### PR TITLE
TICK-0008

### PR DESCRIPTION
**Pull request by DanMedl**
Add Tests to DTO'S Category and Products
Please review and advice. Thanks. 

UPD. I will read more about JUnit tests and think where else I can add them to the classes. (Category and Product)

### CODE REVIEW

**Comment by Shaff3004** [comment]
Where is commit message?

File: com.atb/pom.xml [hunk 1]
```diff
@@ -4,14 +4,22 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <groupId>com.atb</groupId>
-    <artifactId>com.atb</artifactId>
+    <groupId>org.example</groupId>
+    <artifactId>com.this</artifactId>
     <version>1.0-SNAPSHOT</version>
```
Comment (Shaff3004) [inline com.atb/pom.xml:9]: Why did you change it?

File: com.atb/src/test/java/com/atb/dtos/CategoryTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+    @Test(expected = InvalidCodeException.class)
+    public void shouldThrowInvalidCodeException_whenCodeIsEmpty() {
+        category.setCode("");
+    }
+
+    @Test
+    public void shouldSet_WhenCodeIsValid() {
+        category.setCode("C0001");
```
Comment (Shaff3004) [inline com.atb/src/test/java/com/atb/dtos/CategoryTest.java:35]: I forgot to mention one crucial point regarding the test:
assertions should be separated with empty line, so your code should look like:
`
category.setCode("C0001");

assertEquals("C0001", category.getCode());
`

File: com.atb/src/test/java/com/atb/dtos/CategoryTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        assertEquals("C0001", category.getCode());
+    }
+
+    @Test(expected = InvalidDescriptionException.class)
+    public void shouldThrowInvalidDescriptionException_whenCodeIsNull() {
```
Comment (Shaff3004) [inline com.atb/src/test/java/com/atb/dtos/CategoryTest.java:40]: when description is null

File: com.atb/src/test/java/com/atb/dtos/CategoryTest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        category.setDescription(null);
+    }
+
+    @Test(expected = InvalidDescriptionException.class)
+    public void shouldThrowInvalidDescriptionException_whenCodeIsEmpty() {
```
Comment (Shaff3004) [inline com.atb/src/test/java/com/atb/dtos/CategoryTest.java:45]: when description is empty

File: com.atb/src/test/java/com/atb/dtos/CategoryTest.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        category.setDescription("Lemon");
+        assertEquals("Lemon", category.getDescription());
+    }
+
+    @Test
+    public void shouldSetSubCategory_WhenSubCategoryIsValid() {
+        List<Category> subCategory = new ArrayList<>();
+        subCategory.add(category);
+        category.setSubCategories(subCategory);
+        assertEquals(subCategory, category.getSubCategories());
```
Comment (Shaff3004) [inline com.atb/src/test/java/com/atb/dtos/CategoryTest.java:60]: Please, separate with empty lines

File: com.atb/src/test/java/com/atb/dtos/CategoryTest.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Test
+    public void shouldSet_WhenDescriptionIsValid() {
+        category.setDescription("Lemon");
+        assertEquals("Lemon", category.getDescription());
+    }
+
+    @Test
+    public void shouldSetSubCategory_WhenSubCategoryIsValid() {
```
Comment (Shaff3004) [inline com.atb/src/test/java/com/atb/dtos/CategoryTest.java:56]: _**w**henSubCategoryIsValid

File: com.atb/src/test/java/com/atb/dtos/ProductTest.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    }
+    @Test(expected = InvalidCodeException.class)
+    public void shouldThrowInvalidCodeException_whenCodeIsNull(){
+        product.setCode(null);
+    }
+    @Test(expected = InvalidCodeException.class)
+    public void shouldThrowInvalidCodeException_whenCodeIsEmpty(){
+        product.setCode("");
+    }
+    @Test
```
Comment (Shaff3004) [inline com.atb/src/test/java/com/atb/dtos/ProductTest.java:24]: Please, separate methods with empty lines

File: com.atb/src/test/java/com/atb/dtos/ProductTest.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    public void shouldSet_WhenCodeIsValid(){
```
Comment (Shaff3004) [inline com.atb/src/test/java/com/atb/dtos/ProductTest.java:25]: shouldSetCode_whenCodeIsValid

File: com.atb/src/test/java/com/atb/dtos/ProductTest.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    @Test(expected = InvalidDescriptionException.class)
+    public void shouldThrowInvalidDescriptionException_whenDescriptionIsNull(){
+        product.setDescription(null);
+    }
+    @Test(expected =InvalidDescriptionException.class)
+    public void shouldThrowInvalidDescriptionException_whenDescriptionIsEmpty(){
+        product.setDescription("");
+    }
+    @Test
+    public void shouldSet_WhenDescriptionIsValid() {
```
Comment (Shaff3004) [inline com.atb/src/test/java/com/atb/dtos/ProductTest.java:38]: shouldSetDescription_whenDescriptionIsValid

File: com.atb/src/test/java/com/atb/dtos/ProductTest.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+        product.setDescription("Orange");
+        assertEquals("Orange", product.getDescription());
+    }
+
+
+
```
Comment (Shaff3004) [inline com.atb/src/test/java/com/atb/dtos/ProductTest.java:44]: Please, remove empty lines

File: com.atb/pom.xml (hunk 1 above)
Comment (DanMedl) [reply com.atb/pom.xml:9]: Replace, my mistake

File: com.atb/src/test/java/com/atb/dtos/CategoryTest.java (hunk 2 above)
Comment (DanMedl) [reply com.atb/src/test/java/com/atb/dtos/CategoryTest.java:35]: done

File: com.atb/src/test/java/com/atb/dtos/CategoryTest.java (hunk 3 above)
Comment (DanMedl) [reply com.atb/src/test/java/com/atb/dtos/CategoryTest.java:40]: done

File: com.atb/src/test/java/com/atb/dtos/CategoryTest.java (hunk 4 above)
Comment (DanMedl) [reply com.atb/src/test/java/com/atb/dtos/CategoryTest.java:45]: done

File: com.atb/src/test/java/com/atb/dtos/CategoryTest.java [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+        category.setDescription(null);
+    }
+
+    @Test(expected = InvalidDescriptionException.class)
+    public void shouldThrowInvalidDescriptionException_whenCodeIsEmpty() {
+        category.setDescription("");
+    }
+
+    @Test
+    public void shouldSet_WhenDescriptionIsValid() {
```
[the remark this answers is not in the document]
Comment (DanMedl) [reply com.atb/src/test/java/com/atb/dtos/CategoryTest.java:50]: done

File: com.atb/src/test/java/com/atb/dtos/CategoryTest.java (hunk 5 above)
Comment (DanMedl) [reply com.atb/src/test/java/com/atb/dtos/CategoryTest.java:60]: done

File: com.atb/src/test/java/com/atb/dtos/CategoryTest.java (hunk 6 above)
Comment (DanMedl) [reply com.atb/src/test/java/com/atb/dtos/CategoryTest.java:56]: done

File: com.atb/src/test/java/com/atb/dtos/ProductTest.java (hunk 7 above)
Comment (DanMedl) [reply com.atb/src/test/java/com/atb/dtos/ProductTest.java:24]: done

File: com.atb/src/test/java/com/atb/dtos/ProductTest.java (hunk 8 above)
Comment (DanMedl) [reply com.atb/src/test/java/com/atb/dtos/ProductTest.java:25]: done

File: com.atb/src/test/java/com/atb/dtos/ProductTest.java (hunk 9 above)
Comment (DanMedl) [reply com.atb/src/test/java/com/atb/dtos/ProductTest.java:38]: done

File: com.atb/src/test/java/com/atb/dtos/ProductTest.java (hunk 10 above)
Comment (DanMedl) [reply com.atb/src/test/java/com/atb/dtos/ProductTest.java:44]: done
